front-end/cypress/e2e/pages/transactionDetailPage.ts

```diff
@@ -1,79 +1,158 @@
-import { currentYear, PageUtils } from './pageUtils';
-
-export type ScheduleFormData = {
-  date_received: Date;
-  memo_code: boolean;
-  amount: number;
-  electionType: string; // electionType and electionYear are composite form data for election_code
-  electionYear: number;
-  election_other_description: string;
-  purpose_description: string;
-  memo_text: string;
-  category_code: string;
-};
-
-export const defaultFormData: ScheduleFormData = {
-  date_received: new Date(currentYear, 4 - 1, 27),
-  memo_code: false,
-  amount: 100.55,
-  electionType: '',
-  electionYear: 0,
-  election_other_description: '',
-  purpose_description: PageUtils.randomString(20),
-  memo_text: PageUtils.randomString(20),
-  category_code: '',
-};
+import { LoanFormData, ScheduleFormData } from '../models/TransactionFormModel';
+import { PageUtils } from './pageUtils';
 
 export class TransactionDetailPage {
-  static enterFormData(formData: ScheduleFormData, readOnlyAmount = false, alias = '') {
+
+
+  static enterDate(dateFieldName: string, dateFieldValue: any, alias = '') {
+    alias = PageUtils.getAlias(alias);
+    PageUtils.calendarSetValue(dateFieldName, dateFieldValue, alias);
+  }
+
+  static enterScheduleFormData(formData: ScheduleFormData, readOnlyAmount = false, alias = '') {
     alias = PageUtils.getAlias(alias);
 
-    PageUtils.calendarSetValue('p-calendar[inputid="date"]', formData['date_received'], alias);
-    if (formData['memo_code']) {
+    if (formData.date_received != undefined) {
+      this.enterDate('p-calendar[inputid="date"]', formData.date_received, alias);
+    }
+    if (formData.memo_code) {
       cy.get(alias).find('p-checkbox[inputid="memo_code"]').click();
     }
     if (!readOnlyAmount) {
-      cy.get(alias).find('input#amount').safeType(formData['amount']);
+      cy.get(alias).find('#amount').safeType(formData['amount']);
+    }
+    if (formData.electionType != undefined) {
+      PageUtils.dropdownSetValue('[inputid="electionType"]', formData.electionType, alias);
+    }
+    if (formData.electionYear != undefined) {
+      cy.get(alias).find('#electionYear').safeType(formData.electionYear);
+    }
+    if (formData.election_other_description) {
+      cy.get(alias).find('#election_other_description').safeType(formData.election_other_description);
+    }
+    if (formData.purpose_description) {
+      cy.get(alias).find('textarea#purpose_description').safeType(formData.purpose_description);
+    }
+    cy.get(alias).find('textarea#text4000').safeType(formData.memo_text);
+    
+    if (formData.category_code != "") {
```

Comment:
In this and other places - might want to use !== instead to avoid unexpected behavior

Reply:
made this update